src/utils/constraints-utils.ts

```diff
@@ -0,0 +1,93 @@
+import * as snarkjs from "snarkjs";
+// @ts-expect-error: No type definitions available for the "r1csfile" package
+import * as r1csfile from "r1csfile";
+// @ts-expect-error: No type definitions available for the "@iden3/binfileutils" package
+import * as binfileutils from "@iden3/binfileutils";
+
+import { LinearCombination, R1CSConstraint } from "../../src/types/utils";
+
+export async function getGroth16ConstraintsNumber(r1csFilePath: string): Promise<number> {
+  return (await snarkjs.r1cs.info(r1csFilePath)).nConstraints;
+}
+
+export async function getPlonkConstraintsNumber(r1csFilePath: string, Fr: any): Promise<number> {
+  const normalize = (lc: LinearCombination) => {
+    Object.keys(lc).forEach((key) => {
+      if (lc[key] == 0n) delete lc[key];
+    });
+  };
+
+  const join = (lc1: LinearCombination, k: bigint, lc2: LinearCombination) => {
+    const res = { ...lc1 };
+    Object.keys(lc2).forEach((s) => {
+      res[s] = res[s] ? Fr.add(res[s], Fr.mul(k, lc2[s])) : lc2[s];
+    });
+    normalize(res);
+    return res;
+  };
+
+  const reduceCoefs = (linearComb: LinearCombination, maxC: number) => {
+    let n = Object.keys(linearComb).filter((s) => s !== "0" && linearComb[s] != 0n).length;
+
+    while (n > maxC) {
+      plonkConstraintsCount++;
+      n--;
+    }
+  };
+
+  const addConstraintSum = (lc: LinearCombination) => {
+    reduceCoefs(lc, 3);
+    plonkConstraintsCount++;
+  };
+
+  const getLinearCombinationType = (lc: LinearCombination) => {
+    let k = Fr.zero;
+    let n = 0;
+
+    Object.keys(lc).forEach((key) => {
+      if (lc[key] !== 0n) {
+        if (key === "0") {
+          k = Fr.add(k, lc[key]);
+        } else {
+          n++;
+        }
+      }
```

Comment:
Shitty nested `if`...

```suggestion
      if (lc[key] === 0n) {
        return;
      }

      if (key === "0") {
        k = Fr.add(k, lc[key]);
      } else {
          n++;
      }
```